app/models/AddressResponse.scala

```diff
@@ -32,14 +32,23 @@ case class AddressResponse(
     addressLine2,
     addressLine3,
     addressLine4,
-    postalCode,
+    postCodeFormatter(postalCode),
     getCountry(countryCode)
   ).flatten
 
   // to be extended to other countries in future
   private def getCountry(code: String): Option[String] =
     if (code == "GB") Some("United Kingdom") else Some(code)
 
+  private def postCodeFormatter(postcode: Option[String]): Option[String] =
+    postcode match {
+      case Some(postcode) =>
+        val tail = postcode.substring(postcode.length - 3)
+        val head = postcode.substring(0, postcode.length - 3)
+        Some(s"$head $tail".toUpperCase)
```

Comment:
Is this right? Lets take an input:
```
AA1 2ZZ
```

This would give us a tail of `2ZZ` and a head of `AA1 `

We'd then format it to `AA1  2ZZ` (note the double space)

Reply:
Hm, i should've noticed that - we should strip it first then, will need to update that in reg too, thats where this is used!

Reply:
Ive gone with a **replaceAll("\\s", "")** to cover any whitespace types